packages/teams-js/src/public/search.ts

```diff
@@ -0,0 +1,125 @@
+import { sendMessageToParent } from '../internal/communication';
+import { registerHandler, removeHandler } from '../internal/handlers';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Allows your application to interact with the host M365 application's search box.
+ * By integrating your application with the host's search box, users can search
+ * your app using the same search box they use elsewhere in Teams, Outlook, or Office.
+ *
+ * This functionality is in Beta.
+ * @beta
+ */
+export namespace search {
+  const onChangeHandlerName = 'search.queryChange';
+  const onClosedHandlerName = 'search.queryClose';
+  const onExecutedHandlerName = 'search.queryExecute';
+
+  /**
+   * This interface contains information pertaining to the contents of the host M365 application's search box
+   */
+  export interface SearchQuery {
```

Comment:
I should have said this earlier, apologies, but please put the @beta tag on every exported function/interface/type in this file. This tag shows up in our developer documentation and is a signal to app developers that anything in here may change in the future without being considered a breaking change. This will make it easier for you and your team to continue iterating in the future